docs/DeveloperGuide.md

```diff
@@ -133,6 +133,76 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+5. The information under the `find` section is parsed and appended to `helpMessage`.
+6. The `find` section heading is parsed and assigned to `helpTitle`. 
+7. The `helpMessage` and `helpTitle` are returned via a `CommandResult`.
+8. `LogicManager#execute(userInput)` returns the `CommandResult` to `MainWindow#executeCommand()`, which sets the help window header to `helpTitle` and the content to `helpMessage`.
+9. The help window is display.
+
+The following sequence diagram shows how the 'help' operation works in this scenario:
+
+![HelpSequenceDiagram2](images/HelpSequenceDiagram2.png)
+
+The following activity diagram summarises what happens when a user executes the help command:
+
+![HelpActivityDiagram](images/HelpActivityDiagram.png)
+
+#### Design consideration:
+
+##### Aspect: Where the information displayed in the help window is retrieved from
+
+* **Alternative 1 (current choice):** Retrieve from user guide document stored in resource folder and packed into `JAR` file. 
+  * Pros: Easy to update when features change, command classes will not be cluttered with a long `helpMessage` string.
+  * Cons: Have to remember to copy the latest version of the user guide from the docs folder into the resources folder, scanning user guide for information each time help is called can be time consuming, have to parse markdown into plain text.
+
+* **Alternative 2:** Retrieve from `helpMessage` string stored in each `Command` class. 
+  * Pros: Easy to implement, easy to retrieve `helpMessage`, minimal processing needed.
+  * Cons: To update the `helpMessage` of a command, we must search for its class and edit the `helpMessage` string manually. This essentially means we have do 2 updates every time a change is made to a command: one to the user guide and one to the command's class. 
+
+* **Alternative 3:** Retrieve from user guide webpage.
+  * Pros: Minimal updating needed when features change since only the user guide in the docs folder needs to be updated.
+  * Cons: Does not work when HeliBook is used offline, implementation might be complicated, scanning user guide for information each time help is called can be time-consuming.
+  
+Alternative 1 was eventually chosen as we were planning to make major changes to HeliBook over several iterations. Since we are already expected to update the user guide with each iteration, it is more efficient to simply copy the latest user guide document into the resources folder after updates are made rather than to edit each `helpMessage`. Furthermore, as long as the format of the user guide remains constant, parsing the markdown text into plain text is manageable and does not take too much time. This alternative will also work when HeliBook is used offline, making the application easy to use on the go. Lastly, alternative 1 keeps the actual code and documentation separate, making it a more logical and organised implementation. As such, that is the alternative that was chosen. 
+
```

Comment:
Perhaps this can be broken into points for easy reading. 
i.e. Reason for current choice: 
* ...
* ...